server/jetstream_api.go

```diff
@@ -418,9 +418,9 @@ const JSApiConsumerListResponseType = "io.nats.jetstream.api.v1.consumer_list_re
 
 // JSApiConsumerGetNextRequest is for getting next messages for pull based consumers.
 type JSApiConsumerGetNextRequest struct {
-	Expires time.Time `json:"expires,omitempty"`
-	Batch   int       `json:"batch,omitempty"`
-	NoWait  bool      `json:"no_wait,omitempty"`
+	Expires int  `json:"expire_ms,omitempty"`
```

Comment:
See comment above.

Reply:
Yeah I realise. We can try change them all? But asking people to state this kind of thing jn nano seconds (duration) is really not very friendly and neither is time.Time

Reply:
Its expressed as a deadline time via RFC spec.. No ns..

Reply:
Sure this one isn’t ns but I mean the problem in the general case here applies to time and duration cases.

Reply:
That is why I did a deadline and not expiration/ttl.

Reply:
If we do "#{unit}" with a default of ms for a number, I think is a reasonable option for configuration options. I am even ready to concede that timestamps can be in ns, and the library provides a way to convert to the native time representation.

Start time for playback will likely be used often - and UNIX time seems a reasonable resolution - if we are doing formatted ISO date strings, etc, then that itself might be another challenge for the different client languages.

Reply:
This should be a duration if possible. But yeah rather time than ns. In JavaScript it’s just the output from toISOString()

Reply:
Let's maybe sync up Monday am and discuss.

Reply:
Ones to consider in addition to this one:

ConsumerConfig.OptStartTime -> unix seconds
ConsumerConfig.AckWait -> int as seconds
ConsmerInfo.Created -> unix nano
StoredMsg.Time -> unix nano
StreamConfig.MaxAge -> seconds 
StreamConfig.Duplicates -> seconds
StreamState.FirstTime -> unix nano
StreamState.LastTime -> unix nano

Reply:
ok this is helpful. We should rationalize sooner rather than later..

Reply:
Some more, but less nb:

TypedEvent.Time -> unix nano
JSRestoreCompleteAdvisory.Start and End -> unix nano
JSSnapshotCompleteAdvisory.Start and End -> unix nano

Reply:
If they are time.Time then that is ok since we use RFC to format and most (I think all) clients can read and parse and will do the right thing with nanos for their platform.

The time.Duration ones are the ones I am concerned about resolving in a consistent manner.

Reply:
> JSRestoreCompleteAdvisory.Start and End -> unix nano
> JSSnapshotCompleteAdvisory.Start and End -> unix nano

Since these are generic operations for a time, possibly UnixTime on those? - I understand in highspeed messages the timestamp is possibly better as nanos, but this is a tool that will report/start/end times. Regardless, this particular setting is not critical.

Reply:
ms is good imo.

We'll want to do some research into clients.  It seems java and .NET prefer ISO-8601, although RFC 3339 can be parsed in other ways.

Reply:
rfc 3339 is poorly supported on some platforms.  tbh it's been problematic on java, and for .NET it's looking like it'll be challenging.  I think it'll be a much lower level of effort in client development using nanos since epoch.

Reply:
So to avoid guesswork about units lets always use nanos like:

```
ConsumerConfig.OptStartTime -> unix nano seconds since epoch
ConsumerConfig.AckWait -> unix nano as duration
ConsmerInfo.Created -> unix nano seconds since epoch
StoredMsg.Time -> unix nano seconds since epoch
StreamConfig.MaxAge -> unix nano as duration
StreamConfig.Duplicates -> unix nano as duration
StreamState.FirstTime -> unix nano seconds since epoch
StreamState.LastTime -> unix nano seconds since epoch
TypedEvent.Time -> unix nano seconds since epoch
JSRestoreCompleteAdvisory.Start and End -> unix nano seconds since epoch
JSSnapshotCompleteAdvisory.Start and End -> unix nano seconds since epoch
```

So all things are always nanos, it's just a choice between durations or timestamps

Reply:
@derekcollison would you be happy if I send a PR with this change?

Reply:
@aricart any concerns with above?

Reply:
Not yet, let's discuss today.